### DISCUSSION TITLE
Tool for mappers generation

### DISCUSSION THREAD

**Discussion by ichalyshev** [riok/mapperly#1004] [Ideas]
Hi your library is awesome! But source generators still not chaining ability. So f.e. WPF models with observable source generator could not be mapped with mapperly. I think that's why Mapster create separate tool (Mapster.Tool) to generate mappers. And they implement brilliant idea to auto generate new type based on other types f.e. dto objects. Sadly Mapster is not so "graceful" as mapperly.

It would be great if mapperly have external tool for generating mappers.

**Comment by latonz** [comment]
Thank you!
Implementing another interface (despite the source generator) would probably need a lot of work. We would like to keep the focus on the source generator implementation. 
What you can try is to isolate the source generators in different assemblies. Eg. extract your mappers into another assembly which gets compiled only after the assembly with the models is compiled.
